testapps/stresstests/src/main/java/com/microsoft/identity/client/stresstests/fragments/StressTestsFragment.java

```diff
@@ -0,0 +1,466 @@
+                SecretKey secretKey = new SecretKeySpec(tempKey.getEncoded(), "AES");
+                AuthenticationSettings.INSTANCE.setSecretKey(secretKey.getEncoded());
+            }
+        } catch (NoSuchAlgorithmException | InvalidKeySpecException | UnsupportedEncodingException exception) {
+            Log.e(TAG + methodName, "Failed to generate secret key", exception);
+            callback.onError("Failed to generate secret key: " + exception.getMessage());
+        }
+
+        PublicClientApplication.create(getContext(), R.raw.msal_config_default, new IPublicClientApplication.ApplicationCreatedListener() {
+            @Override
+            public void onCreated(IPublicClientApplication application) {
+                callback.onSuccess(application);
+            }
+
+            @Override
+            public void onError(MsalException exception) {
+                Log.e(TAG + methodName, "Failed to load MSAL Application", exception);
+                callback.onError(String.format("Failed to load MSAL Application: %s", exception.getMessage()));
+            }
+        });
+    }
+
+
+    private void updateTimer(final long startTime, final long timeLimitMinutes) {
+        timeElapsed = timeElapsed + DELAY;
+        final long timeRemaining = TimeUnit.MINUTES.toMillis(timeLimitMinutes) - timeElapsed;
+
+        handler.post(new Runnable() {
+            @Override
+            public void run() {
+                timeElapsedTextView.setText(Util.timeString(timeElapsed / 1000));
+                timeRemainingTextView.setText(Util.timeString(timeRemaining / 1000));
+            }
+        });
+    }
+
+    public synchronized void printOutput(final String text) {
+        final String currentTime = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss", Locale.US).format(new Date());
+
+        handler.post(new Runnable() {
+            @Override
+            public void run() {
+                resultsTextView.append(String.format("%s: %s\n", currentTime, text));
+                final int length = resultsTextView.getText().length();
+                if (length > 1000) {
+                    resultsTextView.setText(resultsTextView.getText().subSequence(length - 1000, length));
+                }
+                resultsScrollView.fullScroll(View.FOCUS_DOWN);
+            }
+        });
+    }
+
+
+    /**
+     * Provide the title that will be displayed at the top of the layout
+     *
+     * @return the title of the tests running
+     */
+    public abstract String getTitle();
+
+    /**
+     * Provide the number of threads that will be running
+     *
+     * @return the number of threads
+     */
+    public abstract int getNumberOfThreads();
+
+    /**
+     * Provide the time limit for the tests in minutes.
+     *
+     * @return the time limit in minutes
+     */
+    public abstract int getTimeLimit();
+
+    /**
+     * Prepare the fragment for running the tests. The callback's result will be passed to every execution
```

Comment:
Is the behavior specific to `Fragments`? Or is this more like test setup?

Reply:
It's more like test set-up. The results from this step are used in each iteration of the stress test.